### ISSUE TITLE
error when enrolling in study using `npm run watch`: `Core._storePendingConsent - Unable to uninstall ion-basic-study@mozilla.org Error: Could not establish connection. Receiving end does not exist.`

### ISSUE THREAD

**Issue by hamilton** [mozilla-rally/rally-core-addon#556]
It appears that a bug has been introduced into the Core Add-On or something has changed within Firefox. When I click "join study", it takes a second to fetch the study, but I first get this error:

```
Core._storePendingConsent - Unable to uninstall ion-basic-study@mozilla.org Error: Could not establish connection. Receiving end does not exist.
```

It seems that if it takes too long for the study to return any value, we throw an error, even if we should be waiting.

**Comment by rhelmer** [comment]
@Dexterp37 is this related to the side-load detection/prevention you landed in d72dd24da2bd41451e4236f59231d32d60625bb7 ?

**Comment by rhelmer** [comment]
@hamilton I would have thought this message is just informative, is there an actual bug? Does it prevent study install, or studies from sending data etc?

**Comment by Dexterp37** [comment]
> 
> 
> @hamilton I would have thought this message is just informative, is there an actual bug? Does it prevent study install, or studies from sending data etc?

Yeha, +1 on @rhelmer said: how is the experience broken for you? What does this cause?

**Comment by hamilton** [comment]
Right, this breaks the user experience with `npm run watch`, so in this sense it is a bug. When I join a study (click "join study" and then consent in the dialog) I would expect the study to download, and when I install it, there should be a change in the UI that shows I joined the study. I have to refresh the options page to see that. In this sense it is _technically_ doing what it is supposed to, but the error that is being thrown shows that the Core Add-On isn't waiting long enough to verify that the study.

**Comment by hamilton** [comment]
I've uploaded a video of what the experience looks like, as well as the error message as it appears.

![Kapture 2021-04-22 at 12 14 23](https://user-images.githubusercontent.com/95735/115772755-556a0280-a364-11eb-9f07-da5a122ad793.gif)

**Comment by rhelmer** [comment]
Thanks for the screencap! Hm, I am not sure this is related to the error actually... the UI should change after the extension is installed not as soon as "accept" is tapped. I'll take a look into this today.

**Comment by rhelmer** [comment]
@hamilton I was chatting with @Dexterp37 and he pointed out the `Content.dispatcher` TypeError, which was likely fixed by https://github.com/mozilla-rally/rally-core-addon/commit/6a483720435eec1a19efac8ef97288b826b77290

Would you mind re-testing? I'll do so today if I have a moment as well. Thanks for flagging this!

**Comment by rhelmer** [comment]
> @hamilton I was chatting with @Dexterp37 and he pointed out the `Content.dispatcher` TypeError, which was likely fixed by [6a48372](https://github.com/mozilla-rally/rally-core-addon/commit/6a483720435eec1a19efac8ef97288b826b77290)
> 
> Would you mind re-testing? I'll do so today if I have a moment as well. Thanks for flagging this!

This does wfm after update and `npm install` to pick up the latest glean, I'll leave this open for you to verify @hamilton, also we should consider making the message you filed the issue with clearer that it is non-fatal.

**Comment by hamilton** [comment]
@rhelmer just pulled down and installed. The merge appears to have fixed this. Thanks for addressing ~ will close this.
